lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -3343,6 +3347,38 @@ void AtenCatOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// AtenLogOp
+//===----------------------------------------------------------------------===//
+
+void AtenLogOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                            MLIRContext *context) {
+  patterns.add(+[](AtenLogOp op, PatternRewriter &rewriter) {
+    Location loc = op->getLoc();
+    if (op->getNumOperands() != 1) {
+      return failure();
+    }
+    Value self = getScalarIntValue(op->getOperand(0), loc, rewriter);
+    if (!self)
+      self = getScalarFloatValue(op->getOperand(0), loc, rewriter);
+    if (!self) {
+      return failure();
+    }
+    auto outType = op->getResult(0).getType();
+    double floatScalar;
+    int64_t intScalar;
+    double resultScalar;
+    if (matchPattern(self, m_TorchConstantInt(&intScalar)))
+      resultScalar = std::log(intScalar);
+    if (matchPattern(self, m_TorchConstantFloat(&floatScalar)))
+      resultScalar = std::log(floatScalar);
+    Value result = rewriter.create<ConstantFloatOp>(
+        loc, rewriter.getF64FloatAttr(resultScalar));
+    rewriter.replaceOpWithNewOp<PrimNumToTensorScalarOp>(op, outType, result);
```

Comment:
Use `PrimNumToTensorScalarOp` will cause dtype mismatch when `outType`'s element type is `f32`. Because that `PrimNumToTensorScalarOp (!torch.float)` should infer outType to be `f64`.